src/wazuh_modules/wm_control.c

```diff
@@ -80,32 +80,36 @@ char* getPrimaryIP(){
 #if defined __linux__ || defined __MACH__
     cJSON *object;
     if (sysinfo_network_ptr && sysinfo_free_result_ptr) {
-        sysinfo_network_ptr(&object);
-        if (object) {
-            const cJSON *iface = cJSON_GetObjectItem(object, "iface");
-            if (iface) {
-                const int size_ids = cJSON_GetArraySize(iface);
-                for (int i = 0; i < size_ids; i++){
-                    const cJSON *element = cJSON_GetArrayItem(iface, i);
-                    if(!element) {
-                        continue;
-                    }
-                    cJSON *gateway = cJSON_GetObjectItem(element, "gateway");
-                    if(gateway && cJSON_GetStringValue(gateway) && 0 != strcmp(gateway->valuestring,"unkwown")) {
-                        const cJSON *ipv4 = cJSON_GetObjectItem(element, "IPv4");
-                        if (!ipv4) {
+        if (sysinfo_network_ptr(&object) != -1) {
```

Comment:
this should be == 0, if in the future we add several error codes for return this will be evaluated as success.

Reply:
Will do